File: graphs/2849_determine_if_a_cell_Is_reachable_at_a_given_time.js
/**
 * @param {number} sx
 * @param {number} sy
 * @param {number} fx
 * @param {number} fy
 * @param {number} t
 * @return {boolean}
 */
var isReachableAtTime = function(sx, sy, fx, fy, t) {

    const limit = Math.max(Math.abs(sx - fx), Math.abs(sy - fy))

    return t >= limit && !(limit == 0 && t == 1)

};

File: graphs/2849_determine_if_a_cell_Is_reachable_at_a_given_time.go
func isReachableAtTime(sx int, sy int, fx int, fy int, t int) bool {

    xLimit := int(math.Abs(float64(sx - fx)))
    yLimit := int(math.Abs(float64(sy - fy)))
    limit := int(math.Max(float64(xLimit), float64(yLimit)))

    return t >= limit && !(t == 1 && limit == 0)

}